docs/weave/weave-api-spec/03-querying.md

```diff
@@ -0,0 +1,192 @@
+
+- Path: ``/wallets?prefix``, Data: ``0123456789`` (hex) -> db.Iterator(``0123456789``, ``012345678A``)
+  - `wallets` are queried for the accounts starting with from `0123456789` to `012345678A`
+
+## Responses
+
+Since Weave queries routed to tendermint ABCI protocol, tendermint enforces responses to be in `key/value` format. Reference: [tendermint/abci-spec#query](https://tendermint.readthedocs.io/en/v0.21.0/abci-spec.html#query).
+
+```bash
+curl -X POST -d '{ "json-rpc": 2.0, "id": "foobar321", "method": "abci_query", "params": { "path": "/wallets?prefix", "data": "CBC76ADED2C9DB439DB4C8D714CF26" } }' https://bns.davenet.iov.one/
+```
+
+When the curl command above executed, this response will be received:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": "foobar321",
+  "result": {
+    "response": {
+      "key": "ChljYXNoOsvHat7SydtDnbTI1xTPJtrlIpqR",
+      "value": "ChAKAggBEgoIlZrvOhoDSU9W",
+      "height": "2506"
+    }
+  }
+}
+```
+
+Since the request is with prefix, meaning multiple values will be returned.
+
+### Parsing responses
+
+The `key` and `value` values here are encoded with [ResultSet](https://github.com/iov-one/weave/blob/v0.18.0/spec/proto/app/results.proto#L5-L9). In order to decode ResultSet, import protobuf definition as mentioned in [docs/weave/transaction](https://github.com/iov-one/docs/blob/master/docs/weave/weave-api-spec/01-transaction.md#L102). You can take a look at [iov-core](https://github.com/iov-one/iov-core/blob/v0.15.0/packages/iov-bns/src/bnsconnection.ts#L674-L679) implementation.
+
+__Important:__ Every key must include the bucket's name as prefix. As you can see on [iov-core/iov-bns](https://github.com/iov-one/iov-core/blob/v0.15.0/packages/iov-bns/src/bnsconnection.ts#L159-L177) and [iov-core/iov-bns/bnsconnection/getAccount](https://github.com/iov-one/iov-core/blob/v0.15.0/packages/iov-bns/src/bnsconnection.ts#L341) every key field in the ResultSet must begin with `cash:`
+
+## Available bucket paths by package name
+
+- __Username__
+  - [key prefix](https://github.com/iov-one/weave/blob/v0.18.0/cmd/bnsd/x/username/model.go#L73): `username:`
+  - `/usernames` -> takes `id` as data, returns [bnsd/x/username/username.Token](https://github.com/iov-one/weave/blob/master/cmd/bnsd/x/username/codec.proto#L7-L26)
+  - `/usernames?prefix` -> takes prefix as data, returns `ResultSet`
+
+- __Cash__
+  - [key prefix](https://github.com/iov-one/weave/blob/v0.18.0/x/cash/model.go#L18): `cash:`
+  - `/wallets` -> takes `id`(8 bytes), returns [x/cash.Set](https://github.com/iov-one/weave/blob/v0.18.0/x/cash/codec.proto#L11-L14) by id
+  - `/wallets?prefix` -> takes prefix as data, returns `ResultSet`
+
+- __Sigs__
+  - [key prefix](https://github.com/iov-one/weave/blob/v0.18.0/x/sigs/model.go#L117): `sigs:`
+  - `/auth` -> takes address `id`(8 bytes) returns [x/sigs.UserData](https://github.com/iov-one/weave/blob/v0.18.0/spec/proto/x/sigs/codec.proto#L13-L17) by id
```

Comment:
It returns this wrapped in a ResultSet of one item.
I think you comment this above, but the difference between non-prefix and prefxed form here, makes one think it is differnet.

I would assume ResultSet is known and just a say

"retruns UserData by id (or empty result)"
prefix -> returns 1 or more UserData that begin with that prefix